### ISSUE TITLE
Update source maps in calcite-components-react to avoid warnings

### ISSUE THREAD

**Issue by mpayson** [Esri/calcite-components#3974]
### Description

With the latest create-react-app (5.0), building calcite-components-react results in a stream of warnings about invalid source maps. Looking at some of the source maps in the distribution, it appears they point to an `src` folder that's not included which causes the warnings. Here's an example from `dist/components.js.map`:
> {"version":3,"file":"components.js","sourceRoot":"",**"sources":["../src/components.ts"]** ... }

There's a [CRA issue for this](https://github.com/facebook/create-react-app/discussions/11767) about ignoring these warnings since they don't effect the build, however there's also this [relevant comment](https://github.com/facebook/create-react-app/discussions/11767#discussioncomment-1969071):
> The warnings are correct. These packages source maps pointing to src that is not included in the package. The solution would be to remove source maps from the package or add the src at the location the source maps pointing to.

So this issue is to take one of those approaches.

### Acceptance Criteria

`calcite-components-react` doesn't cause warnings in the latest version of create-react-app

### Relevant Info

<details>
<summary>Log excerpt</summary>

```
Compiled with warnings.

Failed to parse source map from '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/components.ts' file: Error: ENOENT: no such file or directory, open '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/components.ts'

Failed to parse source map from '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/index.ts' file: Error: ENOENT: no such file or directory, open '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/index.ts'

Failed to parse source map from '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/createComponent.tsx' file: Error: ENOENT: no such file or directory, open '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/createComponent.tsx'

Failed to parse source map from '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/createOverlayComponent.tsx' file: Error: ENOENT: no such file or directory, open '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/createOverlayComponent.tsx'

Failed to parse source map from '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/index.ts' file: Error: ENOENT: no such file or directory, open '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/index.ts'

Failed to parse source map from '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/utils/attachProps.ts' file: Error: ENOENT: no such file or directory, open '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/utils/attachProps.ts'

Failed to parse source map from '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/utils/case.ts' file: Error: ENOENT: no such file or directory, open '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/utils/case.ts'

Failed to parse source map from '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/utils/index.tsx' file: Error: ENOENT: no such file or directory, open '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/utils/index.tsx'

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

assets by chunk 12.5 MiB (id hint: vendors) 148 assets
50 assets
1643 modules

WARNING in ./node_modules/@esri/calcite-components-react/dist/components.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/components.ts' file: Error: ENOENT: no such file or directory, open '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/components.ts'
 @ ./node_modules/@esri/calcite-components-react/dist/index.js 1:0-29 1:0-29
 @ ./src/App.tsx 8:0-134 51:30-42 66:35-52 71:38-54 74:40-58 76:42-55 87:41-54 98:41-54 114:39-57 115:42-55 125:41-54 169:35-52 174:38-54 176:40-58 178:41-54 193:39-57 194:42-55 206:41-54 215:41-54
 @ ./src/index.tsx 6:0-24 27:39-42

WARNING in ./node_modules/@esri/calcite-components-react/dist/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/index.ts' file: Error: ENOENT: no such file or directory, open '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/index.ts'
 @ ./src/App.tsx 8:0-134 51:30-42 66:35-52 71:38-54 74:40-58 76:42-55 87:41-54 98:41-54 114:39-57 115:42-55 125:41-54 169:35-52 174:38-54 176:40-58 178:41-54 193:39-57 194:42-55 206:41-54 215:41-54
 @ ./src/index.tsx 6:0-24 27:39-42

WARNING in ./node_modules/@esri/calcite-components-react/dist/react-component-lib/createComponent.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/createComponent.tsx' file: Error: ENOENT: no such file or directory, open '/Users/maxp8108/devtopia/tempdev/webworker/arcgis-data-pipelines/webapp/node_modules/@esri/calcite-components-react/src/react-component-lib/createComponent.tsx'
 @ ./node_modules/@esri/calcite-components-react/dist/react-component-lib/index.js 1:0-57 1:0-57
 @ ./node_modules/@esri/calcite-components-react/dist/components.js 6:0-61 7:45-65 8:49-69 9:42-62 10:45-65 11:47-67 12:46-66 13:45-65 14:41-61 15:42-62 16:41-61 17:48-68 18:42-62 19:40-60 20:44-64 21:40-60 22:47-67 23:55-75 24:53-73 25:44-64 26:48-68 27:53-73 28:46-66 29:49-69 30:51-71 31:57-77 32:44-64 33:49-69 34:48-68 35:39-59 36:42-62 37:40-60 38:41-61 39:42-62 40:40-60 41:50-70 42:41-61 43:51-71 44:48-68 45:51-71 46:41-61 47:40-60 48:40-60 49:44-64 50:49-69 51:42-62 52:41-61 53:42-62 54:42-62 55:47-67 56:46-66 57:41-61 58:44-64 59:49-69 60:48-68 61:43-63 62:50-70 63:44-64 64:47-67 65:52-72 66:46-66 67:50-70 68:42-62 69:41-61 70:42-62 71:41-61 72:50-70 73:46-66 74:42-62 75:48-68 76:47-67 77:43-63 78:47-67 79:42-62 80:39-59 81:42-62 82:44-64 83:40-60 84:40-60 85:46-66 86:51-71 87:46-66 88:39-59 89:44-64 90:46-66 91:43-63 92:50-70 93:40-60 94:44-64 95:45-65 96:49-69
 @ ./node_modules/@esri/calcite-components-react/dist/index.js 1:0-29 1:0-29
 @ ./src/App.tsx 8:0-134 51:30-42 66:35-52 71:38-54 74:40-58 76:42-55 87:41-54 98:41-54 114:39-57 115:42-55 125:41-54 169:35-52 174:38-54 176:40-58 178:41-54 193:39-57 194:42-55 206:41-54 215:41-54
 @ ./src/index.tsx 6:0-24 27:39-42
```

</details>

### Which Component

calcite-components-react library

**Comment by benelan** [comment]
The react output target has a peer dep of react 16.7.0 which will cause issues with CRA 5 even if this is resolved.

**Comment by benelan** [comment]
> IIRC you can specify the CRA version `npx create-react-app@4 app-name`

Never mind it looks like they don't let you do that anymore
![image](https://user-images.githubusercontent.com/10986395/150585803-d1d5f12b-ec77-4060-88ce-2773ecd9912b.png)

**Comment by mpayson** [comment]
I believe create react app works with older versions of react (by changing the react version in package.json, [stack](https://stackoverflow.com/questions/46566830/how-to-use-create-react-app-with-an-older-react-version) & [CRA](https://github.com/facebook/create-react-app/discussions/9917) discussions), so it's possible to meet the peer dependency and still have the issue. Also, I think this stems from using [webpack-contrib/source-map-loader](https://github.com/webpack-contrib/source-map-loader#ignoring-warnings) so it may show up in other areas?

**Comment by benelan** [comment]
I'm pretty sure you have to downgrade `react-scripts` too which would be downgrading CRA. Do you have a repro sample of `react-scripts` v5x and `react` v16x?

**Comment by benelan** [comment]
@mpayson, this should be fixed in the coming `0.17.0` release of `@esri/calcite-components-react`. Can you please verify using your app after the release? Thanks!

**Comment by mpayson** [comment]
Verified in my app, closing as resolved, thank you!
